src/vellum/workflows/nodes/bases/base.py

```diff
@@ -293,15 +293,17 @@ def should_initiate(
 
             if cls.merge_behavior == MergeBehavior.AWAIT_ALL:
                 """
-                A node utilizing an AWAIT_ALL merge strategy will only be considered ready for the Nth time
-                when all of its dependencies have been executed N times.
+                A node utilizing an AWAIT_ALL merge strategy will only be considered ready
+                when all of its dependencies have invoked this node.
                 """
-                current_node_execution_count = state.meta.node_execution_cache.get_execution_count(cls.node_class)
-                return all(
-                    state.meta.node_execution_cache.get_execution_count(dep) == current_node_execution_count + 1
+                # Check if all dependencies have invoked this node
+                all_deps_invoked = all(
+                    dep in state.meta.node_execution_cache._dependencies_invoked.get(node_span_id, set())
```

Comment:
[nit] looks like we could initialize `state.meta.node_execution_cache._dependencies_invoked.get(node_span_id, set())` once outside of this loop